### PR TITLE
Finish linting on py3ish code

### PR DESCRIPTION
**Pull request by agjohnson**
This merges in #2863 and #2938, and fixes the rest of the linting problems introduced with #2819. Most of these errors with ordering of imports. The problem with using `future`, and the includes `builtins` with `future`, is that these both cause a number of linting problems. The standard library mocking needs to be at the bottom of the import block, otherwise all the following import statements are considered out of order.

This work normalizes the import order to (mostly) follow pep8:

* Python builtins
* Third party (including `builtins`, which is third party even though it overrides standard imports
* Local packages
* Standard library overrides with `future`, each import needs a `# noqa` to avoid errors on not being able to find the import (as it's just monkey patched via `future`)

This also updated our travis config to test with the correct Python version.

Most important here are the last of the commits to this branch, following d2599ba. The other work is available in other PRs.

### CODE REVIEW

**Comment by agjohnson** [comment]
I think i got all the calls that future was overriding, but it might be a good idea to have @gthank look at the imports that were shuffled. The only thing I don't like about `future` is it's magical and it's not explicit about what is being monkeypatched -- six, although verbose, was clear in its magic.

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
- - TOX_ENV=py27
- - TOX_ENV=py36
- - TOX_ENV=docs
- - TOX_ENV=lint
- - TOX_ENV=eslint
+matrix:
+  include:
+    - python: 2.7
+      env: TOXENV=docs
+    - python: 2.7
```
Comment (ericholscher) [inline .travis.yml]: Is lint different on 2 or 3, should we be linting both languages?

Comment (agjohnson) [inline .travis.yml]: Good question. I was going to say bump linting and docs steps up to py3 when our code is ready. I don't know if bumping up before then would yield any unwarranted effects or not from the code not being quiet py3 compatible.

Comment (ericholscher) [inline .travis.yml]: Makes sense.

**Pull request merged**